src/librustc_resolve/error_reporting.rs

```diff
@@ -409,7 +415,35 @@ impl<'a> Resolver<'a> {
             (Def::Union(..), _) |
             (Def::Variant(..), _) |
             (Def::Ctor(_, _, CtorKind::Fictive), _) if ns == ValueNS => {
-                err.span_label(span, format!("did you mean `{} {{ /* fields */ }}`?", path_str));
+                    match source {
```

Comment:
This is duplicated code, could be extracted to closure/method.

Reply:
Yes, that would be nice.

Reply:
Also, 3 branches out of 4 report exactly the same `err.span_label(...)`, perhaps it can be compacted.